docs/hybrid-slurm-cluster/hybrid-slurm.md

```diff
@@ -0,0 +1,423 @@
+some of the variables defined in the ansible playbook that installs the
+required software for the images.
+
+The variables that are most relevant for the [slurm hybrid module][hybridmodule]
+are listed below:
+
+* [Slurm version][slurmversion]: Packer variable for setting a custom slurm
+  version.
+* [`service_account_scopes`]: Packer variable for attaching service account
+  scopes to the cloud compute VM instances.
+* [`munge_user`]: Ansible role variable for the munge user and group IDs.
+* [`slurm_user`]: Ansible role variable for the slurm user and group IDs.
+
+Remember to update your blueprint to include the `source_image_project` and
+`source_image_family` in the settings under your partitions. As an example, the
+partition definition below sets the `source_image_project` to the `project_id`
+in deployment variables and is using the default `source_image_family` for an
+image created with slurm 21.08.8:
+
+```yaml
+- id: compute-partition
+  source: community/modules/compute/schedmd-slurm-gcp-v5-partition
+  use:
+  - network1
+  settings:
+    partition_name: compute
+    node_count_dynamic_max: 20
+    source_image_project: $(vars.project_id)
+    source_image_family: schedmd-v5-slurm-21-08-8-hpc-centos-7
+```
+
+[slurmgcppacker]: https://github.com/SchedMD/slurm-gcp/tree/v5.1.0/packer
+[example.pkrvars.hcl]: https://github.com/SchedMD/slurm-gcp/tree/v5.1.0/packer/example.pkrvars.hcl
+[slurmversion]: https://github.com/SchedMD/slurm-gcp/blob/v5.1.0/packer/variables.pkr.hcl#L97
+[`service_account_scopes`]: https://github.com/SchedMD/slurm-gcp/blob/v5.1.0/packer/variables.pkr.hcl#L166
+[`munge_user`]: https://github.com/SchedMD/slurm-gcp/blob/v5.1.0/ansible/roles/munge/defaults/main.yml#L17
+[`slurm_user`]: https://github.com/SchedMD/slurm-gcp/blob/v5.1.0/ansible/roles/slurm/defaults/main.yml#L31
+
+## On Premise Setup
+
+### Networking
+
+A VPN must be configured between the on-premise slurm cluster and the Google
+Cloud project. For more information on setting up a Cloud VPN, see the
+[Cloud VPN documentation][cloudvpn].
+
+A DNS service must be configured between the on-premise slurm cluster and the
+Google Cloud project. For more information on setting up a Cloud DNS, see the
+[Cloud DNS documentation][clouddns].
+
+[cloudvpn]: https://cloud.google.com/network-connectivity/docs/vpn/concepts/overview
+[clouddns]: https://cloud.google.com/dns/docs/overview
+
+### Install dependencies
+
+#### Python
+The [hybrid slurm module][hybridmodule] utilizes the
+[Slurm power saving][powersaving] features by setting the resume and suspend
+Programs to python scripts that create and delete cloud compute nodes
+respectively. Because of this, python is required, as are a set of pip packages.
+
+The required pip packages can be installed on the controller by running the
+following command against the [requirements.txt] file in this directory:
+
+```shell
+sudo pip install -r docs/hybrid-slurm-cluster/requirements.txt
```

Comment:
Are these really all needed on the on prem controller? Somehow I thought we got out of that by running terraform on a different machine.

Reply:
They are. Since the scripts are executed on the controller itself, in the very least the google python api package is needed, but since the scripts also use packages like addict, those also need to be there.

It would be great if they could be cut down a bit, but unless we move away from python and towards a complicated bash script, at least one pip package will always be needed.